launch/launch/actions/execute_local.py

```diff
@@ -580,8 +640,11 @@ async def __execute_process(self, context: LaunchContext) -> None:
                     timeout=self.__respawn_delay
                 )
             if not self.__shutdown_future.done():
-                context.asyncio_loop.create_task(self.__execute_process(context))
+                context.asyncio_loop.create_task(
+                    self.__execute_process(context))
                 return
+        if self.__signal_lingering_subprocesses_value:
+            await self._signal_subprocesses(context)
         self.__cleanup()
```

Comment:
Does this mean subprocess cleanup only happens if it isn't restarted? (I'm assuming the `create_task()` call is about recursively calling this coroutine to restart the application?

It seems to me that we should wait for all subprocesses to finish before relaunching the executable no?

Reply:
> Does this mean subprocess cleanup only happens if it isn't restarted? (I'm assuming the create_task() call is about recursively calling this coroutine to restart the application?
>
> It seems to me that we should wait for all subprocesses to finish before relaunching the executable no?

Yes that sounds fine, I didn't think much what should happen when respawning, I will take another look.